src/main/kotlin/Connection.kt

```diff
@@ -36,6 +38,7 @@ class Connection(
 
     fun connect() {
         client.onEvent(this::handleEvent)
+        serverName = client.serverState.serverName
```

Comment:
This can change during connection (it starts as whatever you connected to, then gets replaced when the server tells us its own name). Probably want to update it in ServerReady at some point.